common/rpc/interceptor/telemetry.go

```diff
@@ -469,7 +474,31 @@ func (ti *TelemetryInterceptor) getWorkflowTags(
 		}
 		return workflowTags
 	}
-	return []tag.Tag{}
+
+	// Requests to historyservice and matchingservice have original request field.
+	// It varies in name but type name always ends with "Request".
+	reqV := reflect.ValueOf(req)
```

Comment:
this feels like kind of a hot path to use reflection, especially with a loop and string matching.. maybe we should do this with a code-generated type switch instead?

Reply:
Code generation would be definitely better. I will try to rewrite this.